pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2523,7 +2523,7 @@ public boolean isAllowAutoTopicCreation(final String topic) {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if (EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME.equals(topicName.getLocalName())
+        if (SystemTopicClient.isSystemTopic(topicName)
```

Comment:
`TRANSACTION_BUFFER_SNAPSHOT` was previously not included in this check. I think it is likely preferred to allow all system topics to be auto created.